app/models/seat.rb

```diff
@@ -1,4 +1,5 @@
 class Seat < ApplicationRecord
   belongs_to :bus
+  belongs_to :ticket, optional: true
```

Comment:
can we use the `has_one` here? and what is the difference between using `has_one` and `belongs_to` with `optional: true` or with `optional: false` ?

Reply:
has_one required foreign key in the associated model's table. but we are storing ticket_id as foreign key in Seat model. Thats the reason I'm using belongs_to instead of has_one. We are generating seats while creating bus so we have to set Ticket foreign key to nil as ticket will be generated later . thats why I use optional: true which gives ability to set foreign key to nil.

please correct me If i'm wrong.  Thank you.

Reply:
you are right!